### PR TITLE
Python: Reduce duplicated alerts for flask sources

### PR DESCRIPTION
**Pull request by RasmusWL**
This should help remove "duplicate alerts" from referencing `from flask import request` multiple times within the same function. This should not be a very invasive change, since it will only remove some existing alerts (and should not remove all existing sources within a single function, so therefore also no alerts).

Draft PR to check whether any tests fails, before requesting review ;)

### CODE REVIEW

**Review by yoff — CHANGES_REQUESTED** [review]
This ends up saying: Give me all uses which are neither in an import nor downstream (by local flow) of another source. If one of these nodes has local flow to another, then they should have a common `LocalSourceNode`. I know that you have already shown that such a local source may not be in `potentialFlaskRequestSource` (because it is an import in a local scope) but perhaps it is the right source anyway?

File: python/ql/lib/semmle/python/frameworks/Flask.qll [hunk 1]
```diff
@@ -330,18 +330,53 @@ module Flask {
...
+      //     if cond:
+      //         user_controlled = request.args["foo"]
+      //     else:
+      //         user_controlled = request.args["bar"]
+      //     SINK(user_controlled)
+      // ```
+      //
+      // For a test of this, see
+      // python/ql/test/library-tests/frameworks/flask/SourceTest.ql
+      not DataFlow::localFlowStep(potentialFlaskRequestSource(), this)
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python/ql/lib/semmle/python/frameworks/Flask.qll:379]: good point. I think I did this due to some other attempts to solve this, where I wanted it to be a single step :shrug:

**Comment by RasmusWL** [comment]
Started evaluation to check that we don't loose any taint based on this, but only the duplicated sources. (also checking perf of course)

**Review by yoff — DISMISSED** [review]
Ok, I think this is a good patch for now, but I would like to understand the underlying problem, so we do not have to do this everywhere...

**Comment by RasmusWL** [comment]
Evaluation needs a bit more investigation, so marking this as draft

**Comment by RasmusWL** [comment]
closing in favor of https://github.com/github/codeql/pull/10629

**Pull request closed without merging**